README.rst

```diff
@@ -26,3 +26,371 @@ Run `tox` to run the whole test suite:
+
+Users of RIOT and `FIT IoT-LAB <https://www.iot-lab.info/>`__ may have
+already ran experiments on multiple ctrls of the same type (e.g:
+``iotlab-m3``) using the ``IOTLAB_NODE`` make environment variable. With
+this one can easily control which device it is targeting.
+
+But if running this locally, with e.g.: multiple ``samr21-xpro``
+connected the serial or ``DEBUG_ADAPTER_ID`` must be used to flash the
+correct device, and for some ``BOARD``\ s also the serial port ``PORT``.
+These variables can be appended to the environment of the spawned
+object, e.g:
+
+-  `FIT IoT-LAB <https://www.iot-lab.info/>`__:
+
+.. code:: python
+
+   # first device using dwm1001-1 on the saclay site
+   env1 = {'BOARD': 'dwm10001', 'IOTLAB_NODE': 'dwm1001-1.saclay.iot-lab.info'}
+   ctrl1 = RIOTCtrl(env=env1, application_directory='.')
+   # second device using dwm1001-2 on the saclay site
+   env2 = {'BOARD': 'dwm10001', 'IOTLAB_NODE': 'dwm1001-2.saclay.iot-lab.info'}
+   ctrl2 = RIOTCtrl(env=env2, application_directory='.')
+
+-  locally:
+
+.. code:: python
+
+   # first samr21-xpro
+   env1 = {'BOARD': 'samr21-xpro', 'DEBUG_ADAPTER_ID': 'ATML2127031800004957'}
+   ctrl1 = RIOTCtrl(env=env1, application_directory='.')
+   # second samr21-xpro
+   env2 = {'BOARD': 'samr21-xpro', 'DEBUG_ADAPTER_ID': 'ATML2127031800011458'}
+   ctrl2 = RIOTCtrl(env=env2, application_directory='.')
+
+For the advanced user one could also do as suggested in
+`multiple-boards-udev <https://api.riot-os.org/advanced-build-system-tricks.html#multiple-boards-udev>`__
+and use an easy to remember variable to identify BOARDs (which would
+allow also running the same python code on different setups), if
+following the above guide:
+
+.. code:: python
+
+   # first samr21-xpro
+   env1 = {'BOARD': 'samr21-xpro', 'BOARD_NUM': 0}
+   ctrl1 = RIOTCtrl(env=env1, application_directory='.')
+   # second samr21-xpro
+   env2 = {'BOARD': 'samr21-xpro', 'BOARD_NUM': 1}
+   ctrl2 = RIOTCtrl(env=env2, application_directory='.')
+
+Factories
+~~~~~~~~~
+
+The same tasks are done multiple times creating the object flashing it,
+starting the terminal and making sure its clean up. Once experiments
+grow and take over multiple ctrls this can become tedious, using a
+Factory together with a context manager can help with this.
+
+Going back to our example lets write a factory inheriting from
+``RIOTCtrlFactoryBase``:
+
+.. code:: python
+
+   from contextlib import ContextDecorator
+   from riotctrl.ctrl import RIOTCtrl, RIOTCtrlBoardFactory
+   from riotctrl_ctrl import native
+
+   class RIOTCtrlAppFactory(RIOTCtrlBoardFactory, ContextDecorator):
+
+       def __init__(self):
+           super().__init__(board_cls={
+               'native': native.NativeRIOTCtrl,
+           })
+           self.ctrl_list = list()
+
+       def __enter__(self):
+           return self
+
+       def __exit__(self, *exc):
+           for ctrl in self.ctrl_list:
+               ctrl.stop_term()
```

Comment:
IMHO this should be part of `RIOTCtrlFactoryBase` (or a terminal-specific extension of it?). Many will use it as such, when documented here, so let's prevent future copy-paste, and put it into the base code.

Reply:
IMO that would be out of scope, can we first get a base README in and the do modifications to RIOTCtrl?